<launch>
  <!-- <arg name="namespace" default="/robot1"/> -->
  <arg name="namespace" default=""/>
  <arg name="run_sim" default="true"/>
  <arg name="run_rviz" default="true"/>
  <arg name="run_joystick" default="false"/>
  <arg name="spawn_gazebo" default="false" />
  <arg name="x_pos" default="-7.0"/>
  <arg name="y_pos" default="-1.0"/>
  <arg name="z_pos" default=" 0.0"/>
  <arg name="yaw"   default=" 1.57"/>
  <arg name="dynamics_model" default="unicycle_dynamics"/>
  <arg name="model" default="$(find avc_urdf)/urdf/bvc.xacro" />
  <arg name="rvizconfig" default="$(find avc_motion)/rviz/urdf.rviz" />
  <arg name="goal_topic" default="/move_base_simple/goal" />
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />

  <!-- Start Simulation -->
  <group if="$(arg run_sim)">
      <include file="$(find gazebo_ros)/launch/empty_world.launch" if="$(arg spawn_gazebo)">
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/turtlebot3_house.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="false"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
      </include>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" if="$(arg run_rviz)"/>

  </group>

  <!-- Create Vehicle -->
  <group>
    <include file="$(find avc_motion)/launch/avc_rvizSim.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="namespace" value="$(arg namespace)" />
      <arg name="spawn_gazebo" value="$(arg spawn_gazebo)" />
      <arg name="x_pos" value="$(arg x_pos)"/>
      <arg name="y_pos" value="$(arg y_pos)"/>
      <arg name="z_pos" value="$(arg z_pos)"/>
      <arg name="yaw" value="$(arg yaw)"/>
    </include>

    <!-- Run the goal to goal control -->
    <node pkg="controllers" type="simple_go_to_goal" name="go_to_goal"  output="screen" unless="$(arg run_joystick)">
        <param name="nominal_velocity" value="0.5" />
        <param name="trans_vel_gain" value="0.5" />
        <param name="rot_vel_gain" value="1.0" />

        <remap from="/cmd_vel" to="$(arg namespace)/cmd_vel" />
        <remap from="goal" to="$(arg goal_topic)" />
        <remap from="odom" to="$(arg namespace)/odom" />
    </node>
  </group>



</launch>
